script.js

```diff
@@ -1 +1,221 @@
-// Please implement exercise logic here
+// GLOBAL VARIABLES
+
+//board size
+let boardSize;
+
+//input for board size
+let boardSizeInput;
+
+//submit button for board size
+let submit;
+
+// keep data about the game in a 2-D array
+let board = [];
+// let board = [
+//   ["", "", ""],
+//   ["", "", ""],
+//   ["", "", ""],
+// ];
+
+// the element that contains the rows and squares
+let boardElement;
+
+// the element that contains the entire board
+// we can empty it out for convenience
+let boardContainer;
+
+// current player global starts at X
+let currentPlayer = "X";
+
+//message display
+let message;
+
+//HELPER FUNCTIONS
+// completely rebuilds the entire board every time there's a click
+const buildBoard = (board) => {
+  // start with an empty container
+  boardContainer.innerHTML = "";
+  boardElement = document.createElement("div");
+  boardElement.classList.add("board");
+
+  // move through the board data array and create the
+  // current state of the board
+  for (let i = 0; i < board.length; i += 1) {
+    // separate var for one row / row element
+    const row = board[i];
+    const rowElement = document.createElement("div");
+    rowElement.classList.add("row");
+
+    // set each square
+    // j is the column number
+    for (let j = 0; j < row.length; j += 1) {
+      // one square element
+      const square = document.createElement("div");
+      square.classList.add("square");
+
+      // set the text of the square according to the array
+      square.innerText = board[i][j];
+
+      rowElement.appendChild(square);
+
+      // set the click all over again
+      // eslint-disable-next-line
+      square.addEventListener("click", () => {
+        squareClick(i, j);
+      });
+    }
+
+    // add a single row to the board
+    boardContainer.appendChild(rowElement);
+  }
+};
+
+//GAME INITIALIZATION LOGIC
+// create the board container element and put it on the screen
+const initGame = () => {
+  boardContainer = document.createElement("div");
+  document.body.appendChild(boardContainer);
+
+  message = document.createElement("p");
+  document.body.appendChild(message);
+
+  boardSizeInput = document.createElement("input");
+  document.body.appendChild(boardSizeInput);
+
+  submit = document.createElement("button");
+  document.body.appendChild(submit);
+  submit.innerHTML = "Submit the board size";
+
+  submit.addEventListener("click", () => {
+    boardSize = Number(boardSizeInput.value);
+    console.log(boardSize);
+    for (let i = 0; i < boardSize; i += 1) {
```

Comment:
good work setting a board of variable size!
consider adding a comment here explaining what your code does